File: ASP_lez03_EF_Manuale_Ferramenta/ASP_lez03_EF_Manuale_Ferramenta/Models/Prodotto.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ASP_lez03_EF_Manuale_Ferramenta.Models
{
    [Table("Prodotto")]
    public class Prodotto
    {
        public int ProdottoId { get; set; }

        public string? Codice { get; set; }

        public string Nome { get; set; } = null!;

        public string Categoria { get; set; } = null!;

        public string? Descrizione { get; set; }

        public decimal Prezzo { get; set; }

        public int? Quantita { get; set; }

        public DateTime? DataCreazione { get; set; }
    }
}

File: TS_Lezioni/ang_02_gestionale/src/app/models/prodotto.ts
export class Prodotto {
    codice: string | undefined;
    nome: string | undefined;
    descrizione: string | undefined;
    prezzo: number | undefined;

    constructor(codice?: string, nome?: string, descrizione?: string, prezzo: number = 0){
        this.codice = codice;
        this.nome = nome;
        this.descrizione = descrizione;
        this.prezzo = prezzo;
    }
}
